### DISCUSSION TITLE
What is the principle of engraving

### DISCUSSION THREAD

**Discussion by MingSongJiang** [InfinyTech3D/SofaUnity#133] [Help - Using SofaUnity] [answered]
I wonder how sofaunity is carved? Does the cutter touch and destroy the corresponding tetrahedron？
I want the carving to penetrate the object faster, now the carving has to wait a long time to cut through the object and needs to move very slowly to cut through the object

**Comment by epernod** [comment, marked as answer]
Hi,

yes the principle of carving is deleting element. So as soon as the cutter/laser touch an element (triangle or tetrahedron) it is deleted and the physical behavior is updated.

This is different from cutting which rely on element subdivision and split. This features is available but still in development as it is more complex. 

Best regards
